### ISSUE TITLE
How do I find out the callers bus name?

### ISSUE THREAD

**Issue by cmcantalupo** [rhinstaller/dasbus#55]
I'm working on adding a dbus service for the the geopm project:

https://github.com/geopm/geopm/pull/1606

I would like to know the Unix user id of the process that originated request.  The documentation on how to do this with the dbus module is here:

https://dbus.freedesktop.org/doc/dbus-python/tutorial.html#finding-out-the-caller-s-bus-name

There is an easy way to get the uid from a DBus interface once you know the unique name of the connecting bus.  Is there an equivalent way to do this within an object published with dasbus?

Thanks,
Chris

**Comment by poncovka** [comment]
Hi, it is not supported, but it shouldn't be too difficult to implement. I will have to think it through, but the result could look like this:
```

@dbus_interface("org.example.HelloWorld"")
class HelloWorld(object):

    @expects_additional_arguments
    def Hello(self, name: Str, **call_arguments) -> Str:

        # Get the bus name that invoked the method.
        sender = call_arguments.get("sender")

        # Return a message.
        return "Hello {}!".format(sender)

```

Give me a week or so.

**Comment by cmcantalupo** [comment]
Thanks @poncovka for offering to implement the feature so quickly!  This is not a blocking issue for me (there is plenty more work to be done on the geopm service that is unrelated), so I can wait a bit for the feature to land.

**Comment by cmcantalupo** [comment]
@poncovka

I have tested the PR you made with the feature:

https://github.com/rhinstaller/dasbus/pull/57

in my branch of geopm to provide the service:

https://github.com/cmcantalupo/geopm/blob/geopm-service/service/geopmdpy/service.py

and preliminary testing shows that it is working as expected.    

Thanks,
Chris

**Comment by poncovka** [comment]
Thanks for testing!

**Comment by poncovka** [comment]
The fix was released in `dasbus-1.5`.
